packages/training/python/src/training/ab_testing.py

```
@@ -0,0 +1,383 @@
+            lines.append("ARCHETYPE BREAKDOWN:")
+            for archetype, metrics in self.archetype_results.items():
+                lines.append(f"  {archetype}:")
+                lines.append(f"    A: {metrics.get('a_score', 0):.3f}, B: {metrics.get('b_score', 0):.3f}")
+        
+        # Winner determination
+        lines.append("")
+        if self.model_b_wins > self.model_a_wins:
+            improvement = (self.model_b_avg_score - self.model_a_avg_score) / max(abs(self.model_a_avg_score), 0.001) * 100
+            lines.append(f"WINNER: Model B (+{improvement:.1f}% improvement)")
+        elif self.model_a_wins > self.model_b_wins:
+            lines.append(f"WINNER: Model A (baseline)")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)